### PR TITLE
Fix CWL "failed with null" error message

### PR DESCRIPTION
**Pull request by aednichols**
Ran into this while working on #4119 

Before:
```
Workflow input processing failed:
running cwltool on file /var/folders/g_/4k204pzx6sg6gcnqc2bqfykxx8k880/T/e0ae0c88-17c0-476b-9b74-bc5562b22ea3.temp.4377309718155227727/e0ae0c88-17c0-476b-9b74-bc5562b22ea3.cwl failed with null
```
After:
```
Workflow input processing failed:
running cwltool on file /var/folders/g_/4k204pzx6sg6gcnqc2bqfykxx8k880/T/c8eb79b9-eee9-4796-b374-4841ff18a27d.temp.3969925185320342713/c8eb79b9-eee9-4796-b374-4841ff18a27d.cwl failed with Traceback (most recent call last):
  File "/Users/anichols/Library/Caches/Coursier/v1/https/broadinstitute.jfrog.io/broadinstitute/libs-release/org/broadinstitute/heterodon/1.0.0-beta1/heterodon-1.0.0-beta1-single.jar/Lib/heterodon/__init__.py", line 24, in apply
  File "<string>", line 1, in <module>
  File "<string>", line 12, in cwltool_salad
  File "/Users/anichols/Library/Caches/Coursier/v1/https/broadinstitute.jfrog.io/broadinstitute/libs-release/org/broadinstitute/heterodon/1.0.0-beta1/heterodon-1.0.0-beta1-single.jar/Lib/cwltool/load_tool.py", line 279, in validate_document
  File "/Users/anichols/Library/Caches/Coursier/v1/https/broadinstitute.jfrog.io/broadinstitute/libs-release/org/broadinstitute/heterodon/1.0.0-beta1/heterodon-1.0.0-beta1-single.jar/Lib/schema_salad/ref_resolver.py", line 915, in resolve_all
  File "/Users/anichols/Library/Caches/Coursier/v1/https/broadinstitute.jfrog.io/broadinstitute/libs-release/org/broadinstitute/heterodon/1.0.0-beta1/heterodon-1.0.0-beta1-single.jar/Lib/schema_salad/ref_resolver.py", line 1087, in validate_links
schema_salad.validate.ValidationException: ../../../../../var/folders/g_/4k204pzx6sg6gcnqc2bqfykxx8k880/T/c8eb79b9-eee9-4796-b374-4841ff18a27d.temp.3969925185320342713/c8eb79b9-eee9-4796-b374-4841ff18a27d.cwl:11:1: checking field `steps`
../../../../../var/folders/g_/4k204pzx6sg6gcnqc2bqfykxx8k880/T/c8eb79b9-eee9-4796-b374-4841ff18a27d.temp.3969925185320342713/c8eb79b9-eee9-4796-b374-4841ff18a27d.cwl:19:3:   checking object `../../../../../var/folders/g_/4k204pzx6sg6gcnqc2bqfykxx8k880/T/c8eb79b9-eee9-4796-b374-4841ff18a27d.temp.3969925185320342713/c8eb79b9-eee9-4796-b374-4841ff18a27d.cwl#compile`
../../../../../var/folders/g_/4k204pzx6sg6gcnqc2bqfykxx8k880/T/c8eb79b9-eee9-4796-b374-4841ff18a27d.temp.3969925185320342713/c8eb79b9-eee9-4796-b374-4841ff18a27d.cwl:20:5:     Field `run` contains undefined reference to `file:///var/folders/g_/4k204pzx6sg6gcnqc2bqfykxx8k880/T/c8eb79b9-eee9-4796-b374-4841ff18a27d.temp.3969925185320342713/arguments.cwl`
../../../../../var/folders/g_/4k204pzx6sg6gcnqc2bqfykxx8k880/T/c8eb79b9-eee9-4796-b374-4841ff18a27d.temp.3969925185320342713/c8eb79b9-eee9-4796-b374-4841ff18a27d.cwl:12:3:   checking object `../../../../../var/folders/g_/4k204pzx6sg6gcnqc2bqfykxx8k880/T/c8eb79b9-eee9-4796-b374-4841ff18a27d.temp.3969925185320342713/c8eb79b9-eee9-4796-b374-4841ff18a27d.cwl#untar`
../../../../../var/folders/g_/4k204pzx6sg6gcnqc2bqfykxx8k880/T/c8eb79b9-eee9-4796-b374-4841ff18a27d.temp.3969925185320342713/c8eb79b9-eee9-4796-b374-4841ff18a27d.cwl:13:5:     Field `run` contains undefined reference to `file:///var/folders/g_/4k204pzx6sg6gcnqc2bqfykxx8k880/T/c8eb79b9-eee9-4796-b374-4841ff18a27d.temp.3969925185320342713/tar-param.cwl`
```
I'm assuming `getMessage` does not work as expected here because the underlying `Throwable` comes from Python.

I did try `t.value.getMessage` to potentially trim down the stack trace, but that's also `null`.

### CODE REVIEW

File: cwl/src/test/scala/cwl/CwlDecoderSpec.scala [hunk 1]
```diff
@@ -21,10 +22,13 @@ class CwlDecoderSpec extends FlatSpec with Matchers {
...
-      decodeCwlFile(rootPath / "brokenlinks.cwl").
-        value.
-        unsafeRunSync.
-        isLeft shouldBe true
+    val result: Either[NonEmptyList[String], Cwl] = decodeCwlFile(rootPath / "brokenlinks.cwl").
+      value.
+      unsafeRunSync
+
+    result.isLeft shouldBe true
+    "Field `run` contains undefined reference to.+wrong.cwl".r.findAllIn(result.left.get.head).size shouldBe 1
```
Comment (cjllanwarne) [inline cwl/src/test/scala/cwl/CwlDecoderSpec.scala]: uh, is there not a nicer `should contain string`?

Comment (aednichols) [reply cwl/src/test/scala/cwl/CwlDecoderSpec.scala]: Not that I could see that's compatible with regex.

The message is actually
```
Field `run` contains undefined reference to `file:///var/folders/g_/4k204pzx6sg6gcnqc2bqfykxx8k880/T/c8eb79b9-eee9-4796-b374-4841ff18a27d.temp.3969925185320342713/arguments.cwl`
```
where the temporary directory changes every time

File: cwl/src/main/scala/cwl/CwlDecoder.scala [hunk 2]
```diff
@@ -21,7 +21,7 @@ object CwlDecoder {
     val cwlToolResult =
       Try(CwltoolRunner.instance.salad(path))
         .toEither
-        .leftMap(t => NonEmptyList.one(s"running cwltool on file ${path.toString} failed with ${t.getMessage}"))
+        .leftMap(t => NonEmptyList.one(s"running cwltool on file ${path.toString} failed with ${t.value}"))
```
Comment (kshakir) [inline cwl/src/main/scala/cwl/CwlDecoder.scala]: What is `.value`? Could this line be simplified to `s"running cwltool on file $path failed with $t"`?

Comment (aednichols) [reply cwl/src/main/scala/cwl/CwlDecoder.scala]: Good call, not sure why I didn't try that

File: cwl/src/main/scala/cwl/CwlDecoder.scala [hunk 3]
```diff
@@ -21,7 +21,7 @@ object CwlDecoder {
     val cwlToolResult =
       Try(CwltoolRunner.instance.salad(path))
         .toEither
-        .leftMap(t => NonEmptyList.one(s"running cwltool on file ${path.toString} failed with ${t.getMessage}"))
+        .leftMap(t => NonEmptyList.one(s"running cwltool on file ${path.toString} failed with $t"))
```
Comment (kshakir) [inline cwl/src/main/scala/cwl/CwlDecoder.scala]: While you're here, remove the redundant `.toString` then LGTM.

**Pull request merged**